NumLib/ODESolver/ConvergenceCriterion.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_CONVERGENCECRITERION_H
+#define NUMLIB_CONVERGENCECRITERION_H
+
+#include <memory>
+#include "NumLib/NumericsConfig.h"
+
+namespace BaseLib {
+class ConfigTree;
+}  // BaseLib
+
+namespace NumLib
+{
+/*! Convergence criterion for nonlinear solvers.
+ *
+ * It is able to check the difference of the solution vector between iterations
+ * and the residual of the nonlinear equation system.
+ */
+class ConvergenceCriterion
```

Comment:
I would name it `NonlinearConvergenceCriterion` because you explain this is for nonlinear solvers. We need a convergence criterion for coupling iterations.